app/uk/gov/hmrc/accessibilitystatementfrontend/parsers/YamlParser.scala

```diff
@@ -16,21 +16,34 @@
 
 package uk.gov.hmrc.accessibilitystatementfrontend.parsers
 
-import io.circe._
-import io.circe.{yaml => circeYaml}
+import io.circe.{yaml => circeYaml, _}
+import play.api.Logging
+import uk.gov.hmrc.accessibilitystatementfrontend.config.StatementSource
+import scala.util.Try
 
-import scala.io.Source
-
-class YamlParser[T: Decoder] {
+class YamlParser[T: Decoder] extends Logging {
   def parse(yaml: String): Either[Error, T] =
     circeYaml.parser
       .parse(yaml)
       .flatMap(_.as[T])
 
-  def parseFromSource(source: Source): Either[Error, T] =
+  def parseFromSource(statementSource: StatementSource): Either[Throwable, T] =
     try {
-      parse(source.mkString)
-    } finally {
-      source.close()
+      logger.info(s"Parsing YAML source file for source: ${statementSource.filename}")
+
+      val parsedYaml: Either[Throwable, T] =
+        Try(statementSource.source.mkString)
```

Comment:
Will this still close the source if there is an error? Is it worth writing a test for this?

Reply:
Yes it will, ignore me :-)